### ISSUE TITLE
Virtual DOM not working in element with "flex: auto"

### ISSUE THREAD

**Issue by hknutzen** [olifolkerd/tabulator#1779]
**Describe the bug**
I know, that Tabulator needs a fixed height to render a table with virtual DOM.
But if the containing element gets its height by "flex: 1 1 auto" AND if the outer element has only "min-height" but no "height" set, then virtual DOM doesn't work correctly:
1. table.rowManager.renderMode is in fact set to "virtual", but
2. height of containing element is enlarged to fit all rows, so
3. effectively, no virtual DOM is applied.

**Tabulator Info**
version 4.1.4

**To Reproduce**
1. Go to https://jsfiddle.net/xLjc925s/2/
2. Look at output: Table is rendered outside of screen
3. Look at log messages: clientHeight of element changes, after tabulator was rendered

**Expected behavior**
Tabulator should not change height of its containing element.
Only that many rows should be rendered, that fit into screen.

**Additional context**
The outer div has "min-height: 100vh;". If we change that to "height: 100vh;" the bug disappears.
This is a stripped down example of  a Vue + Vuetify layout. For some reason, there the outer div only gets "min-height" set. So I need this fixed to make Tabulator usable inside flexible Vuetify layouts.

**Comment by olifolkerd** [comment]
Hey @hknutzen 

Please ask questions on stack overflow, this list is reserved for bug reports and feature requests.

This is not a bug with Tabulator. the issue is the way you have configured your layout.

The Tabulator element must have a height set on it otherwise it cannot enable the virtual DOM which enable classic render mode. in classic render mode the table will render without a scrollbar and will expand its height to fit all rows.

In your case you want to put an additional element inside your "inner" element to contain the table, and set the height on this to 100%

```html
<div class="outer">
  <div>Header</div>
  <div class="inner red-box">
       <div id="example-table"></div>
  </div>
  <div>Footer</div>
</div>
```

Cheers

Oli :)

**Comment by hknutzen** [comment]
Hey Oli,

I think you have not looked close enough into my bug report.
1. The inner element has got a value for "clientHeight".
   Exactly "this.table.element.clientHeight" is checked  by row_manager.js at line 1117.
2. The attribute table.rowManager.renderMode is set to "virtual".
   So your code actually tried to enable the virtual DOM mode.

Please look again at https://jsfiddle.net/xLjc925s/2/
The console shows log messages for
1. clientHeight
2. renderMode

Thanks
Heinz

**Comment by olifolkerd** [comment]
Tabulator is not designed to run in an element resized by flex. My advice stands. You should use it on an element inside the flex box.

**Comment by matkuki** [comment]
Hey @olifolkerd ,

Sorry for posting in this issue, but I only just found this issue now and I think I have a similar problem that I also posted on `StackOverflow`, with no responses: https://stackoverflow.com/questions/63881278/rendering-when-inserting-lots-of-rows

The [`Flexx`](https://github.com/flexxui/flexx) library that I'm trying to embed `Tabulator` into uses `flex` boxes everywhere, so when I found this issue I tried this:
```
<div id="test-table-outer">
  <div id="test-table"></div>
</div>
```
with the `test-table-outer` being the wrapper and initializing with:
```js
var table = new Tabulator("#test-table", { ...
```
Now all the visual stuff works great (scrolling, drawing, ...), but when inserting a lot of rows one by one (in my test case 100 rows), the lower rows are not rendered, as shown in the `StackOverflow` animation. Only when scrolling down they get rendered, but when scrolling back up they again disappear.

Any help on how to fix would be greatly appreciated.

Thanks,
Matic
